frontend/apps/reader/modules/readerbookmark.lua

```diff
@@ -1012,6 +1014,25 @@ function ReaderBookmark:updateBookmark(item)
     end
 end
 
+
+function ReaderBookmark:patchBookmark(item)
+    for i=1, #self.bookmarks do
+        if item.datetime == self.bookmarks[i].datetime and item.page == self.bookmarks[i].page then
+            local bookmark_before = util.tableDeepCopy(self.bookmarks[i])
+            local is_auto_text = self:isBookmarkAutoText(self.bookmarks[i])
+            for key, val in pairs(item.updated_highlight) do
+                self.bookmarks[i][key] = val
+            end
+            if is_auto_text then
+                self.bookmarks[i].text = self:getBookmarkAutoText(self.bookmarks[i])
+            end
+            self.ui:handleEvent(Event:new("BookmarkUpdated", self.bookmarks[i], bookmark_before))
+            self:onSaveSettings()
```

Comment:
I think both of these are not needed for this migration step.
The event is for Page browser, and mostly only needed when bounds are updated so thumbnails can be trashed.
And onSaveSetting will not really save anything, and we'd rather have it all saved only when it's done, not while it is being done.